design/how-ironic-works.md

```diff
@@ -0,0 +1,356 @@
+      * [What connectivity is required?](#what-connectivity-is-required)
+      * [How-to](#how-to)
+         * [How to discover hardware?](#how-to-discover-hardware)
+         * [How to add hardware to ironic?](#how-to-add-hardware-to-ironic)
+      * [Updating information about a hardware node in ironic.](#updating-information-about-a-hardware-node-in-ironic)
+         * [How do I identify the current state?](how-do-i-identify-the-current-state)
+         * [How to inspect hardware?](#how-to-inspect-hardware)
+         * [How to deploy?](#how-to-deploy)
+         * [How to unprovision a baremetal node?](#how-to-unprovision-a-baremetal-node)
+         * [How to delete a baremetal node?](#how-to-delete-a-baremetal-node)
+         * [How to create the record of an active node](#how-to-create-the-record-of-an-active-node)
+      * [References](#references)
+<!--te-->
+
+This document explains how to use ironic in order to achieve various
+tasks such as creating a node, recreating a node, unprovisioning a
+node, deleting a node.
+
+This is not intended to be design specific documentation, but intends
+to help convey the mechanics of how, such that the reader does not
+have to become an expert in Ironic in order to learn to leverage it.
+
+## How ironic controls hardware
+
+Ironic is largely designed around the ability to issue commands to a
+remote Baseboard Management Controller (BMC) in order to assert the
+desired next boot device and control system power state.
+
+BMCs are also not required, however lacking BMC functionality, then
+power and potential boot mode changes need to be performed by an
+external entity such as a human or a network attached power distribution
+unit.
+
+- Access to the BMC can be via a routed IP network,
+  however this may be less desirable from a security standpoint.
+
+- Some vendor driver interfaces may expect to assert configuration
+  to a BMC where the BMC is expected to reach controller resources
+  or hosted services.
+  Mainly this should be expected when virtual media is in use, which means
+  some sort of dedicated BMC network access becomes desirable, as long as
+  the connectivity matches the failure domain of the machines upon which
+  ironic will be executing.
+
+## How ironic boots hardware
+
+Typically harware is booted utilizing PXE. In the most ideal scenario this
+would be a hybrid PXE/iPXE configuration such that the deployment ramdisk
+for ironic is able to be quickly and efficently transferred to the
+baremetal machine.
+
+In order to assert this configuration at boot time, a dedicated DHCP server
+on semi-dedicated "provisioning" network should be leveraged. This network
+may be reused by other services and systems, but can also be leveraged in
+the case of hardware discovery. The most important factor is that this
+network does not have an external DHCP server attached.
```

Comment:
Ironic provides (or at least manages) a DHCP server, right? And that serves as the "dedicated DHCP server"?

Reply:
Ironic does not manage it directly. We have hooks in the code, but they are all built around neutron. The dedicated DHCP server really just needs to convey to the dhcp client where to obtain the initial bootloader components (like ipxe or grub2) from.